example/calendar.cpp

```diff
@@ -0,0 +1,166 @@
+
+// Copyright (c) 2023 Jiri Nytra (jiri.nytra at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+
+#include <flux.hpp>
+
+#include <algorithm>
+#include <charconv>
+#include <map>
+#include <chrono>
+#include <cstddef>
+#include <functional>
+#include <iostream>
+#include <iterator>
+#include <stdexcept>
+#include <string>
+#include <utility>
+
+using namespace std::chrono;
+using namespace std::string_literals;
+using ymd = year_month_day;
+
+constexpr auto day_pad_size = 1UL;
+constexpr auto day_str_size = day_pad_size + 2UL + day_pad_size;
+constexpr auto week_str_size = 7UL * day_str_size;
+constexpr auto max_weeks_in_month = 6;
+constexpr auto col_sep = "  ";
+constexpr auto row_sep = ' ';
+
+auto dates(ymd from, ymd to)
+{
+    return flux::iota(sys_days{from}, sys_days{to});
+}
+
+auto month_num = [](auto d1, auto d2) { 
+    return ymd(d1).month() == ymd(d2).month(); 
+};
+
+auto week_num = [](auto d1, auto d2) { 
+    return weekday(d1).iso_encoding() < weekday(d2).iso_encoding();
+};
+
+auto day_to_string = [](auto d) { 
+    auto day_num = static_cast<unsigned>(ymd(d).day());
+    auto day_str = (day_num < 10 ? " "s : ""s) + std::to_string(day_num);
+    std::string pad_str(day_pad_size, ' ');
+    return pad_str + day_str + pad_str;
+};
+
+auto month_name (auto d) { 
+    constexpr std::array months = {"Jan", "Feb", "Mar", "Apr", "May", "Jun", 
+                                   "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
+    auto year = std::to_string(static_cast<int>(ymd{d}.year()));
+    auto month = std::string{months[static_cast<unsigned>(ymd{d}.month()) - 1]};
+    auto title = month + '-' + year;
+    auto pad_size = (week_str_size - title.size()) / 2;
+    auto aligned_title = (std::string(pad_size, ' ') + title);
+    aligned_title.resize(week_str_size, ' ');
+    return aligned_title;
+};
+
+auto week_to_string = [](auto&& w) { 
+    auto week_str = flux::from(FLUX_FWD(w))
```

Comment:
I don't think this needs the extra `flux::from` here -- we should just be able to say 

```cpp
FLUX_FWD(w)
    .map(day_to_string)
    .flatten()
    .template to<std::string>();
```